controllers/bsl.go

```diff
@@ -151,14 +151,20 @@ func (r *DPAReconciler) updateBSLFromSpec(bsl *velerov1.BackupStorageLocation, d
 	if err != nil {
 		return err
 	}
-
+	registryDeployment := "True"
+	if bslSpec.Provider == "azure" {
+		if len(bslSpec.Config["storageAccountKeyEnvVar"]) == 0 {
```

Comment:
This looks great, please just add a comment here explaining why we are doing this

Reply:
Added a blurb. Let me know if that makes sense or not